contracts/stake/src/state.rs

```diff
@@ -314,11 +337,13 @@ impl StakeState {
             );
         }
 
-        if stake.reward == 0 {
+        // Shift eligibility only if warnings are saturated
+        if effective_faults > 0 {
             // stake.amount can be None if the provisioner unstake in the same
             // block
             if let Some((_, eligibility)) = stake.amount.as_mut() {
-                *eligibility = next_epoch(rusk_abi::block_height()) + EPOCH;
+                let to_shift = effective_faults * EPOCH;
+                *eligibility = next_epoch(rusk_abi::block_height()) + to_shift;
```

Comment:
Can you add a comment to clarify that the stake is suspended for the rest of the current epoch plus `effective_faults` epochs?

Reply:
Done